### PR TITLE
(#2958, #2981, #2983, #2985) Deprecate features we plan to remove / replace after 1.x

### LINKED ISSUE #2958: Deprecate WebPi source type and source runner
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

WebPi, and the feed, have been retired. The feed is also empty.

image

image

### Describe The Solution. Why is it needed?

We should remove the `webpi` source. As it no longer works, I don't feel we need to deprecate it but open to challenge.

### Additional Context

_No response_

### Related Issues

_No response_

### LINKED ISSUE #2981: Deprecate usage of direct path to nupkg or nuspec in choco install command
The following functionality is being deprecated in v1.3.0 and will be removed in a future version of Chocolatey CLI:

```
choco install C:\Packages\mypackage.nupkg
choco install C:\Packages\unpacked\unpacked.nuspec
```

Instead of specifying a full nuspec or nupkg path, users can specify a package name and `--source=path/to/containing/directory` in order to install locally available packages. For example:

```
choco install mypackage --source=C:\packages
```

Any "loose" packages / bare`.nuspec` files should be first packed with `choco pack` before attempting to install them. For example:

```
choco pack C:\Packages\unpacked\unpacked.nuspec
```

### LINKED ISSUE #2983: Deprecate the --remove option in the apikey command
This will be replaced by remove sub command in future version.

### LINKED ISSUE #2985: Deprecate choco config list including output from features/sources/apikeys etc
Starting in Chocolatey CLI v1.3.0, we are deprecating portions of the `choco config list` output. Users will be warned that the `features` and `sources` sections will no longer be shown when calling `config list` in future versions of Chocolatey, in addition to the API keys not being shown.

To list features or sources, instead use `choco feature` or `choco source` commands, respectively.

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

Adds deprecation messages for:

- WebPI source runner
- `choco install path\to\file.nupkg` / `file.nuspec`
- `config list` showing the features/sources
- `choco apikey`'s `--remove` option.

## Motivation and Context

We plan to remove or replace this functionality in v2.0, so adding deprecation messages now so folks can be aware.

## Testing

Ran through unit tests and then Pester tests in the repo w/ Invoke-Tests.ps1.

### Operating Systems Testing

N/A, just adding log messages and some [Obsolete] attributes here and there.

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [x] Documentation changes.
* [x] PowerShell code changes.

## Change Checklist

* [x] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [x] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue

Fixes #2958 
Fixes #2981 
Fixes #2983 
Fixes #2985

### CODE REVIEW

File: src/chocolatey/infrastructure.app/domain/SourceTypes.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System;
+
 namespace chocolatey.infrastructure.app.domain
 {
     /// <summary>
```
Comment (corbob) [inline src/chocolatey/infrastructure.app/domain/SourceTypes.cs:21]:
```suggestion
namespace chocolatey.infrastructure.app.domain
{

    using System;

    /// <summary>
```

Chocolatey convention is for using statements to go inside the namespace.

File: tests/chocolatey-tests/commands/choco-config.Tests.ps1 [hunk 2]
```diff
@@ -125,6 +125,15 @@ Describe "choco config" -Tag Chocolatey, ConfigCommand {
             $Output.Lines | Should -Contain "Use choco apikey to interact with API keys."
         }
 
+        It "Displays deprecation note about sources and features" {
+            $pattern = @(
+                "Starting in v2.0.0, the `config list` command will only list the"
+                " configuration values, instead of also listing the configured features"
+                " and sources."
+            ) -join "\r?\n"
+            $Output.String | Should -Match "Starting in v2.0.0, the `config list` command will only list the\r?\n configuration values, instead of also listing the configured features\r?\n and sources."
```
Comment (corbob) [inline tests/chocolatey-tests/commands/choco-config.Tests.ps1:134]: Is there a reason this sets up a pattern and then doesn't use it? Also, correct me if I'm wrong, but doesn't `Invoke-Choco` trim the lines before joining them onto the `String` property?

**Review by corbob — CHANGES_REQUESTED** [review]

File: tests/chocolatey-tests/commands/choco-config.Tests.ps1 (hunk 2 above)
Comment (vexx32) [reply tests/chocolatey-tests/commands/choco-config.Tests.ps1:134]: Huh, oops. Let me double check on that and update this.

Comment (vexx32) [reply tests/chocolatey-tests/commands/choco-config.Tests.ps1:134]: image
Invoke-Choco currently only trims lines for `.Lines`, not `.String` it seems

File: src/chocolatey/infrastructure.app/domain/SourceTypes.cs (hunk 1 above)
Comment (vexx32) [reply src/chocolatey/infrastructure.app/domain/SourceTypes.cs]: Fixed!

**Review by corbob — APPROVED** [review]
LGTM!

File: src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs [hunk 3]
```diff
@@ -542,9 +542,9 @@ public override void Because()
             }
 
             [Fact]
-            public void should_not_output_warnings()
+            public void should_output_deprecation_warnings()
             {
-                MockLogger.Messages.Keys.ShouldNotContain("Warn");
+                MockLogger.Messages.Keys.ShouldContain("Warn");
```
Comment (AdmiringWorm) [inline src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs:547]: Should probably also check that it outputs the correct deprecation warning, not just that it outputs one.

File: src/chocolatey/infrastructure.app/domain/SourceType.cs [hunk 4]
```diff
@@ -26,7 +26,7 @@ public enum SourceType
     {
         //this is what it should be when it's not set
         normal,
-
+        [Obsolete("The WebPI service has been retired, and this source type will be removed in v2.0.0 of Chocolatey.")]
```
Comment (AdmiringWorm) [inline src/chocolatey/infrastructure.app/domain/SourceType.cs:29]: You don't need to have this, the entire enumeration is already deprecated in favor of the `SourceTypes` class instead. Although, there should not be any harm in keeping it though.

**Review by AdmiringWorm — APPROVED** [review]
Just a couple of comments, other than those these changes looks good to me.

File: src/chocolatey/infrastructure.app/domain/SourceType.cs (hunk 4 above)
Comment (vexx32) [reply src/chocolatey/infrastructure.app/domain/SourceType.cs:29]: Figured I'd put it in for consistency to avoid confusion for now yeah.

**Review by corbob — APPROVED** [review]
Looks Good To Me. I'll let Test Kitchen finish, then I'll get this merged in.

**Pull request merged**